### PR TITLE
Laying the foundations

### PR DESCRIPTION
**Pull request by danielkz**
Just created the project's skeleton.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+*.ear
+*.zip
+*.tar.gz
+*.rar
+
+# virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
+hs_err_pid*
+
+.idea
+*.iml
```
Comment (mmnaseri) [inline .gitignore]: Add a new line character to the end of all files, always. I think you can even configure your editor to do so for you.

File: summer-build/pom.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    <artifactId>summer-build</artifactId>
+    <packaging>pom</packaging>
+    <version>1.0-SNAPSHOT</version>
+    <modules>
+        <module>../summer-parent</module>
+        <module>../summer-core</module>
+    </modules>
+
+
+</project>
```
Comment (mmnaseri) [inline summer-build/pom.xml]: Add a new line.

File: summer-core/pom.xml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        <groupId>com.danielkz.projects.summer</groupId>
+        <version>1.0-SNAPSHOT</version>
+        <relativePath>../summer-parent/pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>summer-core</artifactId>
+
+
+</project>
```
Comment (mmnaseri) [inline summer-core/pom.xml]: Add a new line.

File: summer-parent/pom.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.danielkz.projects.summer</groupId>
+    <artifactId>summer-parent</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+    
+</project>
```
Comment (mmnaseri) [inline summer-parent/pom.xml]: Add a new line.

File: summer-build/pom.xml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.danielkz.projects.summer</groupId>
+    <artifactId>summer-build</artifactId>
+    <packaging>pom</packaging>
+    <version>1.0-SNAPSHOT</version>
```
Comment (mmnaseri) [inline summer-build/pom.xml]: Do not use `-SNAPSHOT` versions. Let's do proper versioning. Start at `0.1.0` and let's move onward from there.

We are following [semantic versioning](https://semver.org/spec/v2.0.0.html).

File: summer-parent/pom.xml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.danielkz.projects.summer</groupId>
+    <artifactId>summer-parent</artifactId>
+    <version>1.0-SNAPSHOT</version>
```
Comment (mmnaseri) [inline summer-parent/pom.xml]: Use semantic versioning. Version this as `0.1.0`. Once you update this, do not forget to update the references (i.e. `summer-core/pom.xml`).

**Comment by danielkz** [comment]
Nice! ok sure. Will remember them all.

**Pull request merged**